services/QuillLMS/client/app/bundles/PremiumHub/components/dataExportTableAndFields.tsx

```diff
@@ -181,20 +202,39 @@ export const DataExportTableAndFields = ({ queryKey, searchCount, selectedGrades
     })
   }
 
+  function filtersMatchHash(hashMessage) {
+    const filterTarget = [].concat(
+      queryKey,
+      selectedTimeframe,
+      customTimeframeStartString,
+      customTimeframeEndString,
+      selectedSchoolIds,
+      selectedGrades,
+      selectedTeacherIds,
+      selectedClassroomIds
+    )
+
+    const filterHash = hashPayload(filterTarget)
+
+    return hashMessage == filterHash
+  }
+
+  function customTimeframeMatches(timeframe) {
+    if (!customTimeframeStart || !customTimeframeEnd) return true
+
+    const remoteStart = timeframe?.custom_start?.split('T', 1)[0]
```

Comment:
You might consider a helper method here for repeated logic, e.g. `format_iso_string(string)`

Reply:
Good suggestion.  Working on such a helper earlier in development led to me realizing I didn't need this code at all, and this function in now unused.  (Another point in favor of what you mentioned about too many moving parts in the fix, meaning this didn't get cleaned up.)